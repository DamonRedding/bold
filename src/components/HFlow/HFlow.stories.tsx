import { number, select } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import React from 'react'

import { Button } from '../Button'
import { AlignItems, JustifyContent } from '../Grid/Grid'
import { Icon } from '../Icon'

import { HFlow } from './HFlow'

const justifyContentOptions: JustifyContent[] = [
  'flex-start',
  'flex-end',
  'center',
  'space-between',
  'space-around',
  'space-evenly',
]

const alignItemsOptions: AlignItems[] = ['flex-start', 'flex-end', 'center', 'baseline', 'stretch']

storiesOf('Components|HFlow', module).add('default', () => (
  <HFlow
    hSpacing={number('hSpacing', 1)}
    alignItems={select('alignItems', alignItemsOptions, 'center')}
    justifyContent={select('justifyContent', justifyContentOptions, 'flex-start')}
  >
    <Button kind='primary'>Button 1</Button>
    <Button>Button 2</Button>
    <Button>Button 3</Button>
    <Icon icon='infoCircleFilled' />
  </HFlow>
))
