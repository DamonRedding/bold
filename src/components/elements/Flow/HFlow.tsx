import React from 'react'

import { Flow, FlowProps } from './Flow'

export interface HFlowProps extends FlowProps {}

export function HFlow(props: HFlowProps) {
  return <Flow direction='horizontal' vSpacing={0} hSpacing={1} {...props} />
}
