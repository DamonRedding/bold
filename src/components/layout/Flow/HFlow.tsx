import React from 'react'

import { withStyles } from '../../../styles'

import { Flow, FlowProps } from './Flow'

export interface HFlowProps extends FlowProps {
}

@withStyles
export class HFlow extends React.PureComponent<HFlowProps> {

    render() {
        return (
            <Flow direction='horizontal' vSpacing={0} hSpacing={1} {...this.props}>
                {this.props.children}
            </Flow>
        )
    }
}
