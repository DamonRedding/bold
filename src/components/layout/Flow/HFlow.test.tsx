import { render } from 'enzyme'
import React from 'react'

import { withTheme } from '../../../test'

import { HFlow } from './HFlow'

it('deve renderizar corretamente', () => {
    const wrapper = render(withTheme(
        <HFlow>
            <span>1</span>
            <span>2</span>
            <span>3</span>
        </HFlow>
    ))
    expect(wrapper).toMatchSnapshot()
})

it('deve aceitar Flow props', () => {
    const wrapper = render(withTheme(
        <HFlow vSpacing={2} hSpacing={1}>
            <span>1</span>
            <span>2</span>
            <span>3</span>
        </HFlow>
    ))
    expect(wrapper).toMatchSnapshot()
})
