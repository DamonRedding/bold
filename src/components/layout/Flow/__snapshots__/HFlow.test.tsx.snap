// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve aceitar Flow props 1`] = `
.emotion-3 {
  margin: 0 -0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-0 {
  margin-top: 1rem;
  margin-right: 0.5rem;
  margin-bottom: 1rem;
  margin-left: 0.5rem;
  display: inline-block;
}

<div
  class="emotion-3"
>
  <div
    class="emotion-0"
  >
    <span>
      1
    </span>
  </div>
  <div
    class="emotion-0"
  >
    <span>
      2
    </span>
  </div>
  <div
    class="emotion-0"
  >
    <span>
      3
    </span>
  </div>
</div>
`;

exports[`deve renderizar corretamente 1`] = `
.emotion-3 {
  margin: 0 -0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-0 {
  margin-top: 0rem;
  margin-right: 0.5rem;
  margin-bottom: 0rem;
  margin-left: 0.5rem;
  display: inline-block;
}

<div
  class="emotion-3"
>
  <div
    class="emotion-0"
  >
    <span>
      1
    </span>
  </div>
  <div
    class="emotion-0"
  >
    <span>
      2
    </span>
  </div>
  <div
    class="emotion-0"
  >
    <span>
      3
    </span>
  </div>
</div>
`;
